src/coreclr/nativeaot/Runtime/GCHelpers.cpp

```diff
@@ -354,71 +356,52 @@ static Array* AllocateNewArrayImpl(Thread* pThread, EEType* pArrayEEType, uint32
         size = ALIGN_UP(size, sizeof(uintptr_t));
     }
 
-    size_t max_object_size;
-#ifdef HOST_64BIT
-    if (g_pConfig->GetGCAllowVeryLargeObjects())
-    {
-        max_object_size = (INT64_MAX - 7 - min_obj_size);
-    }
-    else
-#endif // HOST_64BIT
-    {
-        max_object_size = (INT32_MAX - 7 - min_obj_size);
-    }
-
-    if (size >= max_object_size)
```

Comment:
Is this checked somewhere else?

Reply:
Yup, it's checked in the newly factored out GcAllocInternal. We had all of this duplicated between RhpGcAlloc and AllocateNewArrayImpl. Instead of adding a third copy for AllocateNewObjectImpl, I'm funneling it all to the common helper. But please double check, I would hate to mess this up.